sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/implementation/WebApplicationFirewallPolicyImpl.java

```diff
@@ -0,0 +1,291 @@
+            .findFirst();
+
+        if (ruleSetOptional.isPresent()) {
+            ManagedRuleSet ruleSet = ruleSetOptional.get();
+            // if Bot Protection rule present, only set non-null new version
+            if (version != null) {
+                ruleSet.withRuleSetVersion(version);
+            }
+        } else {
+            ensureManagedRules().managedRuleSets().add(
+                new ManagedRuleSet()
+                    .withRuleSetType(BOT_DETECTION_RULE_SET_TYPE)
+                    .withRuleSetVersion(versionOrDefault));
+        }
+        return this;
+    }
+
+    @Override
+    public WebApplicationFirewallPolicyImpl enableRequestBodyInspection() {
+        ensurePolicySettings().withRequestBodyCheck(true);
+        return this;
+    }
+
+    @Override
+    public WebApplicationFirewallPolicyImpl disableRequestBodyInspection() {
+        ensurePolicySettings().withRequestBodyCheck(false);
+        return this;
+    }
+
+    @Override
+    public WebApplicationFirewallPolicyImpl withRequestBodySizeLimitInKb(int limitInKb) {
+        ensurePolicySettings()
+            .withRequestBodyInspectLimitInKB(limitInKb)
+            .withMaxRequestBodySizeInKb(limitInKb);
+        return this;
+    }
+
+    @Override
+    public WebApplicationFirewallPolicyImpl withFileUploadSizeLimitInMb(int limitInMb) {
+        ensurePolicySettings()
+            .withFileUploadLimitInMb(limitInMb);
+        return this;
+    }
+
+    @Override
+    public Mono<WebApplicationFirewallPolicy> createResourceAsync() {
+        return this.manager().serviceClient().getWebApplicationFirewallPolicies()
+            .createOrUpdateAsync(this.resourceGroupName(), this.name(), this.innerModel())
+            .map(innerToFluentMap(this));
+    }
+
+    @Override
+    public WebApplicationFirewallPolicyImpl withManagedRuleSet(KnownManagedRuleSet managedRuleSet) {
+        Objects.requireNonNull(managedRuleSet);
+        return withManagedRuleSet(managedRuleSet.type(), managedRuleSet.version());
+    }
+
+    @Override
+    public WebApplicationFirewallPolicyImpl withManagedRuleSet(KnownManagedRuleSet managedRuleSet,
+                                                               ManagedRuleGroupOverride... managedRuleGroupOverrides) {
+        Objects.requireNonNull(managedRuleSet);
+        return withManagedRuleSet(managedRuleSet.type(), managedRuleSet.version(), managedRuleGroupOverrides);
+    }
+
+    @Override
+    public WebApplicationFirewallPolicyImpl withManagedRuleSet(String type, String version) {
+        return withManagedRuleSet(type, version, (ManagedRuleGroupOverride[]) null);
+    }
+
+    @Override
+    public WebApplicationFirewallPolicyImpl withManagedRuleSet(String type, String version,
+                                                               ManagedRuleGroupOverride... managedRuleGroupOverrides) {
+        ManagedRuleSet managedRuleSet = new ManagedRuleSet().withRuleSetType(type).withRuleSetVersion(version);
```

Comment:
Do we need
```java
        Objects.requireNonNull(type);
        Objects.requireNonNull(version);
```

Reply:
I think we can let service check them. I'm towards minimal parameter check on client side.

Reply:
If one can set null to `with`, why you do this validation on `without`?

Reply:
Got it... It's originally for convenience in client code, not have to worry about `version.equals()` null pointer. Thanks for catching that, I'll modify the equals comparison.